### PR TITLE
Add AAC decoding for MP4 files.

### LINKED ISSUE #374: Add AAC input support
We need to add AAC decoding in order to add sound support for mp4 files.

### PR DESCRIPTION
**Pull request by jerzywilczek**
This PR uses the FDK deocder for decoding and Symphonia for demuxing. Most MP4s should work in this configuration. The remaining work is:

- allow OPUS decoding for MP4s
- allow AAC decoding for RTP

Closes #374

### CODE REVIEW

File: compositor_pipeline/src/pipeline/decoder/fdk_aac.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    UnsupportedChannelConfig,
+
+    #[error("The aac decoder cannot decode chunks with kind {0:?}.")]
+    UnsupportedChunkKind(EncodedChunkKind),
+}
+
+impl Decoder {
+    fn new(options: AacDecoderOptions) -> Result<Self, AacDecoderError> {
+        // MP4_RAW should cover a huge majority of MP4 AAC streams (95+%), and detecting which transport
+        // is used in the mp4 seems to be very hard
```
Comment (jerzywilczek) [inline compositor_pipeline/src/pipeline/decoder/fdk_aac.rs:26]: It actually isn't that hard, the information is just so scattered in various specifications that it was very hard to find. I know how to do this now and will implement it in a  subsequent PR.

File: compositor_pipeline/src/pipeline/decoder/fdk_aac.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            })
+            .unwrap();
+
+        result_receiver.recv().unwrap()?;
+
+        Ok(Self)
+    }
+}
+
+fn start_decoding(
```
Comment (wkozyra95) [inline compositor_pipeline/src/pipeline/decoder/fdk_aac.rs:168]:
```suggestion
fn run_decoder_thread(
```

I suggest sticking to a naming convention where
- start means that function does sth and exits quickly while sth that was started is running somewhere
- run means that function does sth in a blocking way

**Review by wkozyra95 — APPROVED** [review]

File: compositor_pipeline/src/pipeline/decoder/fdk_aac.rs (hunk 2 above)
Comment (jerzywilczek) [reply compositor_pipeline/src/pipeline/decoder/fdk_aac.rs:168]: Will do that in the subsequent PR

**Review by WojciechBarczynski — COMMENTED** [review]
LGTM

**Pull request merged**